server/src/com/thoughtworks/go/server/cache/GoCache.java

```diff
@@ -194,6 +196,23 @@ public void removeAll(List<String> keys) {
         }
     }
 
+    public void removeCompositeKeyFromParentCache(String key) {
+        if (key.contains(SUB_KEY_DELIMITER)) {
+            String[] parts = StringUtils.splitByWholeSeparator(key, SUB_KEY_DELIMITER);
```

Comment:
`parentKey` cannot be null as that is one of the keys in the map maintained by ehcache, so it would fail at the time of insertion itself. `childKey` would be null if the key used in the cache is "parent!_#$#_!", but then it would lead to other cache related errors as all such composite caches will have the same key. Moreover `StringUtils.splitByWholeSeparator("parent!_#$#_!", "!_#$#_!")` returns 
```
result = {java.lang.String[2]@1305} 
 0 = "parent"
 1 = ""
```
so we should be ok I think.